pkg/controller/manage/manage_controller.go

```diff
@@ -203,6 +127,176 @@ func NewKmeshManageController(client kubernetes.Interface, security *kmeshsecuri
 	}, nil
 }
 
+func handlePodAddFunc(obj interface{}, namespaceLister v1.NamespaceLister, queue workqueue.RateLimitingInterface, security *kmeshsecurity.SecretManager) {
+	pod, ok := obj.(*corev1.Pod)
+	if !ok {
+		log.Errorf("expected *corev1.Pod but got %T", obj)
+		return
+	}
+
+	namespace, err := namespaceLister.Get(pod.Namespace)
+	if err != nil {
+		log.Errorf("failed to get pod namespace %s: %v", pod.Namespace, err)
+		return
+	}
+
+	if !utils.ShouldEnroll(pod, namespace) {
+		if pod.Annotations[constants.KmeshRedirectionAnnotation] != "enabled" {
+			disableKmeshManage(pod, queue, security)
+		}
+		return
+	}
+	enableKmeshManage(pod, queue, security)
+}
+
+func handlePodUpdateFunc(oldObj, newObj interface{}, namespaceLister v1.NamespaceLister, queue workqueue.RateLimitingInterface, security *kmeshsecurity.SecretManager) {
+	oldPod, okOld := oldObj.(*corev1.Pod)
+	newPod, okNew := newObj.(*corev1.Pod)
+	if !okOld || !okNew {
+		log.Errorf("expected *corev1.Pod but got %T and %T", oldObj, newObj)
+		return
+	}
+
+	namespace, err := namespaceLister.Get(newPod.Namespace)
+	if err != nil {
+		log.Errorf("failed to get pod namespace %s: %v", newPod.Namespace, err)
+		return
+	}
+
+	// enable kmesh manage
+	if oldPod.Annotations[constants.KmeshRedirectionAnnotation] != "enabled" && utils.ShouldEnroll(newPod, namespace) {
+		enableKmeshManage(newPod, queue, security)
+	}
+
+	// disable kmesh manage
+	if oldPod.Annotations[constants.KmeshRedirectionAnnotation] == "enabled" && !utils.ShouldEnroll(newPod, namespace) {
+		disableKmeshManage(newPod, queue, security)
+	}
+}
+
+func handlePodDeleteFunc(obj interface{}, security *kmeshsecurity.SecretManager) {
+	pod, ok := obj.(*corev1.Pod)
+	if !ok {
+		tombstone, ok := obj.(cache.DeletedFinalStateUnknown)
+		if !ok {
+			log.Errorf("couldn't get object from tombstone %#v", obj)
+			return
+		}
+		pod, ok = tombstone.Obj.(*corev1.Pod)
+		if !ok {
+			log.Errorf("tombstone contained object that is not a Job %#v", obj)
+			return
+		}
+	}
+
+	if pod.Annotations[constants.KmeshRedirectionAnnotation] == "enabled" {
+		log.Infof("%s/%s: Pod managed by Kmesh is deleted", pod.GetNamespace(), pod.GetName())
+		sendCertRequest(security, pod, kmeshsecurity.DELETE)
+		// We donot need to do handleKmeshManage for delete, because we may have no change to execute a cmd in pod net ns.
+		// And we have done this in kmesh-cni
+	}
+}
+
+func handleNamespaceAddFunc(Obj interface{}, podLister v1.PodLister, queue workqueue.RateLimitingInterface, security *kmeshsecurity.SecretManager) {
```

Comment:
please change the function param to small cased, this is  golang idiomatic

Reply:
> please change the function param to small cased, this is golang idiomatic

done